### PR TITLE
Add CPU Information

### PR DESCRIPTION
**Pull request by zminhquanz**
Show CPU Information and detect 64 bit CPU

### CODE REVIEW

**Comment by JaCzekanski** [comment]
The builds currently fails, but don't worry - it's my fault ;) I updated submodules recently and must have missed something.

**Comment by zminhquanz** [comment]
So , may i recreate pull request ?

**Comment by JaCzekanski** [comment]
I didn't have time to make a proper code review yet. I'll let you know when I'm done.

File: src/platform/windows/gui/help/sysinfo.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        ImGui::Text("Your CPU have 32 bit");
+    ImGui::End();
+}
+
+void SysInfo::displayWindows() {
+    if (sysinfoWindowOpen) sysinfoWindow();
+}
+
+void do_cpuidex(uint32_t regs[4], uint32_t cpuid_leaf, uint32_t ecxval) { __cpuidex((int *)regs, cpuid_leaf, ecxval); }
+void do_cpuid(uint32_t regs[4], uint32_t cpuid_leaf) { __cpuid((int *)regs, cpuid_leaf); }
```
Comment (JaCzekanski) [inline src/platform/windows/gui/help/sysinfo.cpp:25]: `__cpuid` and `__cpuidex` are Windows-only APIs. I don't think using x86 command for detecting hardware is a good idea.

File: src/platform/windows/gui/help/sysinfo.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+CPUInfo cpu_info;
+
+CPUInfo::CPUInfo() { 
+	Detect(); 
+}
+
+// Detects the various cpu features
+void CPUInfo::Detect() { 
+	memset(this, 0, sizeof(*this));
```
Comment (JaCzekanski) [inline src/platform/windows/gui/help/sysinfo.cpp:35]: "memset" is a big no-no in this case. It's the constructor's responsibility to initialize member fields, but field initialization is often enough.

File: src/platform/windows/gui/help/sysinfo.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+enum CPUVendor {
+    VENDOR_INTEL = 0,
+    VENDOR_AMD = 1,
+};
+
+struct CPUInfo {
+    CPUVendor vendor;
+
+    // Misc
+    char cpu_string[0x21];
```
Comment (JaCzekanski) [inline src/platform/windows/gui/help/sysinfo.h:22]: It's C++, you can just use `std::string`.

File: src/platform/windows/gui/help/sysinfo.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    char brand_string[0x41];
+    bool OS64bit;
```
Comment (JaCzekanski) [inline src/platform/windows/gui/help/sysinfo.h:24]: The application knows if it's running in 64bit or 32bit mode (compile time macros or sizeof pointer).

File: src/platform/windows/gui/help/sysinfo.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    bool Mode64bit;
+
+	bool bAtom;
```
Comment (JaCzekanski) [inline src/platform/windows/gui/help/sysinfo.h:27]: Knowing if the app is running under a certain type of processor is useless for the emulator itself.

**Review by JaCzekanski — COMMENTED** [review]
I'm sorry but this PR doesn't solve any problems or improve the emulator itself.
The emulator doesn't rely on any specific CPU features so runtime detection is redundant.

`cpuid` and winapi dependency limits this functionality only to x86/x64 and Windows platform (Avocado tries to be platform/architecture agnostic). 

If hardware detection would be necessary I would prefer using functions in this order:
- Builtin C++ features, STL or portable macros 
- An external library like SDL
- Manual implementation per platform

If the functionality doesn't have a multiplatform solution then one needs to be implemented for all common platforms (Windows, Linux, MacOS, Android, Web/Emscripten in the future). 
I had to do this in only one place (drive detection in file browser) and I try to limit it to the minimum, as maintenance and testing of `#ifdef` code are hard for me (I develop and run Avocado on Mac).

In this case, using SDL would be a good solution (https://wiki.libsdl.org/CategoryCPU), but again - this isn't something that Avocado needs for now.

**Comment by zminhquanz** [comment]
Thank you for my review

**Pull request closed without merging**